test/Syntax/Parser/diags.swift

```diff
@@ -0,0 +1,18 @@
+// REQUIRES: syntax_parser_lib
+// RUN: %swift-syntax-parser-test %s -dump-diags 2>&1 | %FileCheck %s
+
+// CHECK: [[@LINE+2]]:11 Error: consecutive statements on a line must be separated by ';'
+// CHECK: ([[@LINE+1]]:11,[[@LINE+1]]:11) Fixit: ";"
+let number⁚ Int
+// CHECK: [[@LINE-1]]:11 Error: operator with postfix spacing cannot start a subexpression
+
+// CHECK: [[@LINE+1]]:1 Error: expressions are not allowed at the top level
+5 ‒ 5
+
+// CHECK: [[@LINE+2]]:10 Error: expected ',' separator
+// CHECK: ([[@LINE+1]]:9,[[@LINE+1]]:9) Fixit: ","
+if (true ꝸꝸꝸ false) {}
+
+// CHECK: [[@LINE+2]]:7 Note: unicode character '‒' looks similar to '-'; did you mean to use '-'?
```

Comment:
There's only a CHECK for a `Note`, I assume there's also an error emitted here that is not CHECK'ed ?
It would be good to allow ensuring that all diagnostics for the test case have been CHECK'ed, so if something changes in the parser and new diagnostics get emitted, the test case will fail.

I propose for `swift-syntax-parser-test` to print out at the end a summary of the count of diagnostics (e.g. "2 error(s), 1 warnings(s), 1 note(s)") and CHECK for this at the end of the file.

Reply:
Done